docs/user-guide/setting-up-ci-cd.rst

```diff
@@ -0,0 +1,210 @@
+:desc: Set up a CI/CD pipeline to ensure that iterative improvements to your assistant are tested and deployed with minimum manual effort
+
+.. _setting-up-ci-cd:
+
+Setting up CI/CD
+================
+
+Even though developing an assistant is different than developing a traditional
+software application, you should still follow software
+development best practices. Setting up Continuous Integration (CI) and Continuous Deployment (CD) pipeline ensures that incremental updates
+to your bot are improving it, not harming it.
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Overview
+--------
+
+Continous Integration (CI) is the practice of merging in code changes
+frequently and automatically testing changes as they are committed. Continuous
+Deployment (CD) means automatically deploying integrated changes to a staging
+or production environment. Together, they allow for more frequent improvements
+to your assistant with an efficient workflow for testing and deployment.
+
+This guide will cover **what** should go in a CI/CD pipeline, specific to a
+Rasa project. **How** you implement that pipeline is up to you.
+There are many CI/CD tools out there: `GitHub Actions <https://github.com/features/actions>`_,
+`GitLab CI/CD <https://docs.gitlab.com/ee/ci/>`_, `Jenkins <https://jenkins.io/doc/>`_,
+`CircleCI <https://circleci.com/docs/2.0/>`_, etc. It is recommended to select one that 
+works well with your remote Git repository.
+
+
+Continuous Integration (CI)
+---------------------------
+
+Assistants are best improved with frequent, `incremental updates
+<https://rasa.com/docs/rasa-x/user-guide/improve-assistant/>`_.
+No matter how small a change is, you want to be sure that it doesn't introduce
+new problems or negatively impact the performance of your assistant. Most tests are
+quick enough to run on every change. However, you can set some more
+resource-intentsive tests to run only when certain files have been changed or
+when a certain tag is present.
```

Comment:
what is a tag?

Reply:
`...only when certain files have been changed, or when some other indicator is present. For example, in Github, you can label pull requests manually and set the presence of a specific label as a condition for running a test` ?